scheduler/pkg/kafka/dataflow/server.go

```diff
@@ -358,8 +358,14 @@ func (c *ChainerServer) rebalance() {
 			for server, subscription := range c.streams {
 				if contains(servers, server) {
 					msg.Op = chainer.PipelineUpdateMessage_Create
-					if err := c.pipelineHandler.SetPipelineState(pv.Name, pv.Version, pv.UID, pipeline.PipelineCreating, "Rebalance", sourceChainerServer); err != nil {
-						logger.WithError(err).Errorf("Failed to set pipeline state to creating for %s", pv.String())
+					// we do not need to set pipeline state to creating if it is already in terminating state, and we need to delete it
+					if pv.State.Status == pipeline.PipelineTerminating {
```

Comment:
Noted that we can't be in the "Terminated" state here because we only fetch running pipelines initially (with "Terminating" being considered one of the running states).